src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -288,59 +318,77 @@ protected String extractListenerName(String bootstrapServers) {
      *          The 'advertised.listeners' configuration string.
      */
     protected String[] buildListenersConfig(final InspectContainerResponse containerInfo) {
-        final String bootstrapServers = getBootstrapServers();
-        final String bsListenerName = extractListenerName(bootstrapServers);
         final Collection<ContainerNetwork> networks = containerInfo.getNetworkSettings().getNetworks().values();
         final List<String> advertisedListenersNames = new ArrayList<>();
         final StringBuilder kafkaListeners = new StringBuilder();
         final StringBuilder advertisedListeners = new StringBuilder();
+        
+        String bsListenerName = null;
+        String bootstrapServers = null;
+        
+        // Only get bootstrap servers for broker nodes
+        if (this.nodeRole.isBroker()) {
+            bootstrapServers = getBootstrapServers();
+            bsListenerName = extractListenerName(bootstrapServers);
+        }
 
-        // add first PLAINTEXT listener
-        advertisedListeners.append(bootstrapServers);
-        kafkaListeners.append(bsListenerName).append(":").append("//").append("0.0.0.0").append(":").append(KAFKA_PORT).append(",");
-        this.listenerNames.add(bsListenerName);
+        // Only add client listener for broker nodes
+        if (this.nodeRole.isBroker() && bootstrapServers != null && bsListenerName != null) {
```

Comment:
We don't need `bootstrapServers != null` here as if `this.nodeRole.isBroker()` is true, we've set `bootstrapServers` just above.